### ISSUE TITLE
Web push notification

### ISSUE THREAD

**Issue by tg2dex** [puppeteer/puppeteer#3461]
How I can subscribe, handle and click on push notification?

**Comment by aslushnikov** [comment]
I don't think we have a good story for push notifications. We'll wait for more demand and use cases.

So everybody: please upvote the issue to bump its priority and share your usecase (automation? testing?) in comments here.

Thanks!

**Comment by tg2dex** [comment]
Hi, I want testing my push notifications (from site), but i dont know how I cant do that(

**Comment by reznikGit** [comment]
@aslushnikov Can u help me? Any ideas...

**Comment by pk992** [comment]
I was able to receive and extract push notifications from the browser, but not in headless mode - #3432

My use case is logging the notifications from several pages.

**Comment by reznikGit** [comment]
@paul-kl Hi, u have mail or skype? Let's talk, maybe we can help each other. You managed to click on the notification or get data from the onclick method?

**Comment by doerme** [comment]
some problem:
1.how to allow show notifications, when the chrome dialog alert
2.push service not available, when pushManager.subscribe {headless: true}

**Comment by andyevansweather** [comment]
Use case, test automation: notifications for an applicaiton I am developing need to be tested as a user receives a notification to fill in a form, when a user clicks a submit button on a form, that triggers Notification.close() puppeteer should do this as part of the build.

**Comment by cloudever** [comment]
Any update?

**Comment by reznikGit** [comment]
@aslushnikov ?))

**Comment by s-zaj** [comment]
Use case: automating and load testing user subscriptions during back-end development

**Comment by cloudever** [comment]
@aelialper use case: catch push notifications around the internet for spying the ads traffic 👍

**Comment by reznikGit** [comment]
@cloudever u try spying the push ads traffic? Can u gime your mail?

**Comment by millionwords** [comment]
Any update?

**Comment by appier-rolandwang** [comment]
We would like to use this capability to:
1) Automate function test
2) Monitor notification-scheduler services
3) Test notification at large scale

**Comment by roland-wang-appier** [comment]
We would like to use this capability to:
1) Automate function test
2) Monitor notification-scheduler services
3) Test notification at large scale

**Comment by chinchang** [comment]
Use case: We are a push notification service. We want to automate testing if the push notification working on a website with our code installed.

**Comment by micHar** [comment]
Use case: automate backend e2e tests where a process involves receiving a push and then making use of data from its payload

**Comment by bbrtan** [comment]
upvoted .

there is a need for this , there are triggers for notification  and puppeteer will need an  function to handle it beside going thru overridepermission. ( which somehow weirdly does not work at some situation in chronium )

**Comment by TwoAbove** [comment]
Use case: e2e test with web push server

**Comment by wolfgangmeyers** [comment]
Use case: Headless api testing, needing to test integration with FCM push notifications.

**Comment by vilicvane** [comment]
Push notification is something that could break without raising alert for a noticeable amount of time. It would be useful to have this covered with automated test.

**Comment by tg2dex** [comment]
@aslushnikov

**Comment by erlangparasu** [comment]
@wolfgangmeyers +1

**Comment by erlangparasu** [comment]
> Use case: Headless api testing, needing to test integration with FCM push notifications.

@wolfgangmeyers +1

**Comment by brookmg** [comment]
Is there any update on this issue? Looking for a way to test FCM notifications for different topics.

**Comment by jimmywarting** [comment]
pump!

**Comment by chinchang** [comment]
We need this!

**Comment by i4ybrid** [comment]
I'd like this feature - I want to monitor push notifications and send them to another location (Discord)

**Comment by chinchang** [comment]
Meanwhile, we switched to Cypress and this is how we test things - https://engineering.pushowl.com/mocking-web-push-notification-in-cypress/

**Comment by wowkster** [comment]
Are there any updates on this?
